modal/cli/container.py

```diff
@@ -82,21 +83,22 @@ async def exec(
             raise typer.Exit(code=1)
         raise
 
-    exec_failed = False
-    async with handle_exec_input(client, res.exec_id, use_raw_terminal=tty):
-        try:
-            exit_status = await handle_exec_output(client, res.exec_id, on_connect=connecting_status.stop)
-            if exit_status != 0:
-                rich.print(f"Process exited with status code {exit_status}", file=sys.stderr)
-                exec_failed = True
-        except TimeoutError:
-            connecting_status.stop()
-            rich.print("Failed to establish connection to process", file=sys.stderr)
-            exec_failed = True
-
-    if exec_failed:
-        # we don't want to raise this inside the context manager
-        # since otherwise the context manager cleanup doesn't get called
+    on_connect = asyncio.Event()
+    exec_output_task = asyncio.create_task(handle_exec_output(client, res.exec_id, on_connect=on_connect))
+    try:
+        # time out if we can't connect to the server fast enough
+        await asyncio.wait_for(on_connect.wait(), timeout=15)
+        connecting_status.stop()
```

Comment:
Should this be in a `finally:`? It's in both branches

Reply:
I think it needs to be called before we print the error message, so at least for the `TimeoutError` except case, I don't think it can be moved to `finally` (?)